Fenrir/Commands.fs

```diff
@@ -159,3 +165,86 @@ let stringToByte (s: String): byte[] =
             Byte.Parse(s.AsSpan(charIndex, 2), NumberStyles.AllowHexSpecifier, provider = CultureInfo.InvariantCulture)
         )
     | n -> failwithf "String of invalid length %d: %s" n s
+
+let hashOfObjectInTree (tree: TreeBody) (name: String): byte[] =
+    let atom = Array.find (fun a -> a.Name = name) tree
+    atom.Hash
+
+let changeHashInTree (tree: TreeBody) (hash: byte[]) (name: String): TreeBody =
+    let changer (a: TreeAtom) : TreeAtom =
+        match (a.Name = name) with
+            | true -> {Mode = a.Mode; Name = a.Name; Hash = hash}
+            | false -> a
+    Array.map changer tree
+
+let treeBodyToStream (tree: TreeBody) (stream: Stream): unit =
+    let printAtom (a : TreeAtom): unit =
+        stream.Write(ReadOnlySpan<byte>(a.Mode.ToString(CultureInfo.InvariantCulture) |> Encoding.ASCII.GetBytes))
+        stream.WriteByte(' 'B)
+        stream.Write(ReadOnlySpan<byte>(a.Name |> Encoding.ASCII.GetBytes))
+        stream.WriteByte(00uy)
+        stream.Write(ReadOnlySpan<byte>(a.Hash))
+    Array.iter printAtom tree
+
+let changeHashInCommit (commit: CommitBody) (hash: byte[]): CommitBody =
+    {Tree = hash |> byteToString;
+     Parents = commit.Parents;
+     Rest = commit.Rest}
+
+let commitBodyToStream (commit: CommitBody) (stream: Stream): unit =
+    let printParent (a: String): unit =
+        stream.Write(ReadOnlySpan<byte>("parent "B))
+        stream.Write(ReadOnlySpan<byte>(a |> Encoding.ASCII.GetBytes))
+        stream.WriteByte('\n'B)
+
+    stream.Write(ReadOnlySpan<byte>("tree "B))
+    stream.Write(ReadOnlySpan<byte>(commit.Tree |> Encoding.ASCII.GetBytes))
+    stream.WriteByte('\n'B)
+
+    Array.iter printParent commit.Parents
+    stream.Write(ReadOnlySpan<byte>(String.Join('\n', commit.Rest) |> Encoding.ASCII.GetBytes))
+
+type TreeStreams (length: int) =
+    member this.Streams = Array.create (length) (new MemoryStream())
```

Comment:
As an answer to your first question, do this change:
```suggestion
    member this.Streams = Array.init length (fun _ -> new MemoryStream())
```
Another trouble is that you got confused by F# property syntax and used `member this.Streams = …` instead of `member val Streams = …`

`member this.X = …` defined a getter-only property which will be recalculated each time, while `member val X = …` defines a "member value", which is only set once.
```suggestion
    member val Streams = Array.init length (fun _ -> new MemoryStream())
```